aws_lambda_builders/workflows/ruby_bundler/bundler.py

```diff
@@ -0,0 +1,55 @@
+"""
+Wrapper around calls to bundler through a subprocess.
+"""
+
+import logging
+
+LOG = logging.getLogger(__name__)
+
+class BundlerExecutionError(Exception):
+    """
+    Exception raised when Bundler fails.
+    Will encapsulate error output from the command.
+    """
+
+    MESSAGE = "Bundler Failed: {message}"
+
+    def __init__(self, **kwargs):
+        Exception.__init__(self, self.MESSAGE.format(**kwargs))
+
+class SubprocessBundler(object):
+    """
+    Wrapper around the Bundler command line utility, encapsulating
+    execution results.
+    """
+
+    def __init__(self, osutils, bundler_exe=None):
+        self.osutils = osutils
+        if bundler_exe is None:
+            # TODO: Check Windows issues.
+            bundler_exe = 'bundler'
+
+        self.bundler_exe = bundler_exe
+
+    def run(self, args, cwd=None):
+        if not isinstance(args, list):
+            raise ValueError('args must be a list')
+
+        if not args:
+            raise ValueError('requires at least one arg')
+
+        invoke_bundler = [self.bundler_exe] + args
+
+        LOG.debug("executing Bundler: %s", invoke_bundler)
```

Comment:
Do you think adding absolute path helps here? just to show where bundler is coming from.

Reply:
Traditionally wouldn't be needed, but we can add it if that's important.